elide-core/src/main/java/com/yahoo/elide/security/FilterExpressionCheck.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2016, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.security;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.filter.Predicate;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.parsers.expression.FilterExpressionCheckOkVisitor;
+import com.yahoo.elide.security.checks.InlineCheck;
+
+import java.util.Optional;
+
+/**
+ * Check for FilterExpression
+ *
+ * @param <T> Type of class
+ */
+public abstract class FilterExpressionCheck<T> extends InlineCheck<T> {
+
+    public abstract FilterExpression getFilterExpression(RequestScope requestScope);
+
+    @Override
+    public final boolean ok(T object, RequestScope requestScope, Optional<ChangeSpec> changeSpec) {
+        FilterExpression filterExpression = getFilterExpression(requestScope);
+        return filterExpression.accept(new FilterExpressionCheckOkVisitor(object, this));
+    }
+
+    public boolean applyPredicateToObject(T object, Predicate predicate) {
+        Predicate.PathElement path = predicate.getPath().get(predicate.getPath().size() - 1);
+        Class type = path.getType();
+        //If predicate is not related to object, vacuously true.
+        if (!type.isAssignableFrom(object.getClass())) {
```

Comment:
I don't think you need to do this check.  We can safely assume predicates were built correctly given this is our own code.